identity-wallet/src/state/credentials/mod.rs

```diff
@@ -3,28 +3,47 @@ pub mod reducers;
 
 use super::{core_utils::helpers::get_unverified_jwt_claims, FeatTrait};
 use crate::{error::AppError, state::core_utils::DateUtils};
-
 use derivative::Derivative;
 use identity_credential::{sd_jwt_v2::Sha256Hasher, sd_jwt_vc::SdJwtVc};
 use log::info;
-use oid4vc::oid4vci::credential_format_profiles::CredentialFormats;
+use oid4vc::{
+    oid4vc_core::claim_path_pointer::ClaimPathPointer,
+    oid4vci::{
+        credential_format_profiles::CredentialFormats,
+        credential_issuer::credential_configurations_supported::ClaimDescription,
+    },
+};
 use serde::{Deserialize, Serialize};
 use serde_json::json;
 use ts_rs::TS;
 use uuid::Uuid;
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq, TS)]
+#[ts(export, export_to = "bindings/credentials/DisplayClaim.ts")]
+pub struct DisplayClaim {
+    #[ts(type = "Array<string>")]
+    pub path: ClaimPathPointer,
+    pub key: String,
+    #[ts(type = "any")]
+    pub value: serde_json::Value,
+}
+
 /// A credential displayable by the frontend.
 #[derive(Debug, Serialize, Deserialize, Clone, Eq, Derivative, TS, Default)]
 #[derivative(PartialEq)]
 #[ts(export, export_to = "bindings/credentials/DisplayCredential.ts")]
 pub struct DisplayCredential {
     #[derivative(PartialEq = "ignore")]
     pub id: String,
-    #[ts(type = "string")]
+    #[ts(type = "{ format: string }")]
```

Comment:
This results in frontend code like this:

```
{
  format: {
    format: 'dc+sd-jwt'
  }
}
```

Is that intended?

Reply:
Yes. All `CredentialFormats` variants are essentially serialized as objects that always contain a `format` field and conditionally can have extra fields (depending on the concrete type of `C` in:
```rust
pub enum CredentialFormats<C = ()>
where
    C: FormatExtension,
...
```

So, usually we use it in combination with `#[serde(flatten)]` so that the nested fields are added to the root object (in this case `DisplayCredential`) during serialization, e.g.:
```rust
    #[serde(flatten)]
    pub format: CredentialFormats,
```

We cannot do that here unfortunately because the `TS` trait does not work with `#[serde(flatten)]`.